internal/transport/http2_client.go

```diff
@@ -749,6 +749,24 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (*ClientS
 		callHdr = &newCallHdr
 	}
 
+	// Authority specified in the `CallAuthority` CallOption takes the highest
+	// precedence to determine the :authority header. Any value in Host field of
+	// CallHdr is overwritten. But before overriding, we validate the authority
+	// string against the peer certificates and fail the RPC with `UNAVAILABLE`
```

Comment:
`against peer certificates` is only true for TLS creds. So, we probably can omit that here.